### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by mirtagonz**
Add null pointer controls

### CODE REVIEW

File: src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java [hunk 1]
```diff
@@ -147,46 +148,57 @@ public SnapshotIndicators getSnapshotIndicators(Long snapshotId) {
 
         SnapshotIndicators toRet = new SnapshotIndicators();
 
-        SnapshotEconomicEntity originalSnapshot = economicRepository.findOne(snapshotId);
+        SnapshotEconomicEntity originalSnapshot = economicRepository
+                .findOne(snapshotId);
 
-        SurveyEntity survey = surveyRepository.getOne(originalSnapshot.getSurveyDefinition().getId());
-        List<String> indicatorGroup = survey.getSurveyDefinition().getSurveyUISchema().getGroupIndicators();
+        if (originalSnapshot!=null) {
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java]: Can we do here an early return?
```java
if (originalSnapshot == null) {
   return ;
}
```

File: src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java [hunk 2]
```diff
@@ -147,46 +148,57 @@ public SnapshotIndicators getSnapshotIndicators(Long snapshotId) {
...
-            });
+            List<String> order = survey.getSurveyDefinition()
+                    .getSurveyUISchema().getUiOrder().stream()
+                    .filter(field -> indicatorGroup.contains(field))
+                    .collect(Collectors.toList());
 
+            List<SnapshotIndicatorPriority> priorities = priorityService
+                    .getSnapshotIndicatorPriorityList(originalSnapshot.
+                            getSnapshotIndicator().getId());
+            toRet.setIndicatorsPriorities(priorities);
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java]: I think every `toRet.setXXX` should be at the end of the method, where the other setXXX are.

File: src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java [hunk 3]
```diff
@@ -147,46 +148,57 @@ public SnapshotIndicators getSnapshotIndicators(Long snapshotId) {
...
+
+            SurveyData indicators = indicatorMapper
+                    .entityToDto(originalSnapshot.getSnapshotIndicator());
+            List<SurveyData> indicatorsToRet = new ArrayList<>();
+            if (indicatorGroup != null && !indicatorGroup.isEmpty()
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/surveys/services/impl/SnapshotServiceImpl.java]: Can we refactor this if statement in a separate method? And move all the variables that the if block needs to that method.

**Comment by mirtagonz** [comment]
The recommendations were included!!

**Comment by rodrigojv** [comment]
Thank you @mirtagonz

**Pull request merged**